### PR TITLE
Added support for x-rh-identity

### PR DESCRIPTION
**Pull request by bond95**
# Description

Added support for x-rh-identity, but only for `production` mode

Fixes https://github.com/RedHatInsights/insights-results-aggregator/issues/90

## Type of change

- New feature (non-breaking change which adds functionality)

## Testing steps
Set in server configuration `Debug` to `false` and use x-rh-identity header

### CODE REVIEW

**Comment by bond95** [comment]
@tisnik WDYT?

File: server/auth.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	"net/http"
+)
+
+type contextKey string
+
+const (
+	contextKeyUser = contextKey("user")
+)
+
+type Internal struct {
```
Comment (tisnik) [inline server/auth.go]: if this is exported, it needs to have proper docstring

File: server/auth.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	OrgId string `json:"org_id"`
+}
+
+type Identity struct {
```
Comment (tisnik) [inline server/auth.go]: dtto

File: server/server.go [hunk 3]
```diff
@@ -53,6 +54,8 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
+var Environment = os.Getenv("AGGREGATOR_ENV")
```
Comment (tisnik) [inline server/server.go]: I'd urge to use Configuration there, not "package global" variables. Please

File: server/server.go [hunk 4]
```diff
@@ -111,6 +117,16 @@ func (server HTTPServer) readOrganizationID(writer http.ResponseWriter, request
 		return 0, errors.New(message)
 	}
 
+	if identityContext != nil && Environment == "production" {
+		identity := identityContext.(Identity)
+		if Environment == "production" && identity.Internal.OrgId != organizationIDParam {
```
Comment (tisnik) [inline server/server.go]: enum type there?

File: server/auth.go (hunk 1 above)
Comment (tisnik) [reply server/auth.go]: IMHO it might not be exported...?

**Comment by bond95** [comment]
@tisnik @natiiix thank you all for feedback

File: server/server.go (hunk 3 above)
Comment (tisnik) [reply server/server.go]: for example we can add "production bool" into the Configuration. Do the config setup properly in main package and let be server itself basically global-variable free. WDYT?

File: server/server.go (hunk 4 above)
Comment (tisnik) [reply server/server.go]: or better: please add "production bool" into Configuration and change this line to something like: if server.config.production && ...

**Comment by tisnik** [comment]
just please make CI happy :)

**Pull request merged**